charts/envoy/templates/deployment.yaml

```diff
@@ -53,12 +53,25 @@ spec:
             value: {{ include "envoy.fullname" . }}-headless
           - name: service_listeners
             value: "{{ .Values.envoyConfiguration.serviceListeners }}"
+          startupProbe:
+            httpGet:
+              path: /ready
+              port: 9001
+            failureThreshold: 60
+            periodSeconds: 5
```

Comment:
I set the timeout value to 300sec (`failureThreshold * periodSeconds = 300sec`). If the pod won't be `Ready` state after 300sec, Kubernetes terminates the pod.